test_fakelump_repartitioning failing with invalid iterator decrement

There is an iterator error when running tests with debug information.

Info:
```
$ uname -a
Linux rosewater 4.2.5-1-ARCH #1 SMP PREEMPT Tue Oct 27 08:13:28 CET 2015 x86_64 GNU/Linux
$ python --version
Python 3.5.0
$ gcc --version
gcc (GCC) 5.2.0

$ make clean
$ make debug
$ make test
...
tests.test_lump.test_fakelump_repartitioning ... /usr/include/c++/5.2.0/debug/safe_iterator.h:361:error: attempt to 
    decrement a dereferenceable (start-of-sequence) iterator.

Objects involved in the operation:
iterator "this" @ 0x0x7ffe1a50e970 {
type = N11__gnu_debug14_Safe_iteratorISt23_Rb_tree_const_iteratorISt4pairIKyyEENSt7__debug3mapIyySt4lessIyESaIS4_EEEEE (constant iterator);
  state = dereferenceable (start-of-sequence);
  references sequence with type `NSt7__debug3mapIyySt4lessIyESaISt4pairIKyyEEEE' @ 0x0x7ffe1a50eac0
}
```

--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,8 @@
+2015-08-14  Luiz Irber  <khmer@luizirber.org>
+
+   * lib/subset.cc: check iterator before decrementing in
+   repartition_largest_partition
+
 2015-10-25  Titus Brown  <titus@idyll.org>
 
    * scripts/normalize-by-median.py,tests/test_normalize_by_median.py: test

--- a/lib/subset.cc
+++ b/lib/subset.cc
@@ -1431,11 +1431,11 @@ unsigned long long SubsetPartition::repartition_largest_partition(
 
     // find biggest.
     PartitionCountDistribution::const_iterator di = d.end();
-    --di;
 
     if (d.empty()) {
         throw khmer_exception();
     }
+    --di;
 
     for (PartitionCountMap::const_iterator cmi = cm.begin(); cmi != cm.end();
             ++cmi) {
@@ -1450,13 +1450,15 @@ unsigned long long SubsetPartition::repartition_largest_partition(
 #if VERBOSE_REPARTITION
     std::cout << "biggest partition: " << di->first << "\n";
 #endif // 0
-    --di;
 
 #if VERBOSE_REPARTITION
     std::cout << "biggest partition ID: " << biggest_p << "\n";
 #endif // 0
 
-    next_largest = di->first;
+    if (di != d.begin()) {
+        --di;
+        next_largest = di->first;
+    }
 
 #if VERBOSE_REPARTITION
     std::cout << "next biggest partition: " << di->first << "\n";

